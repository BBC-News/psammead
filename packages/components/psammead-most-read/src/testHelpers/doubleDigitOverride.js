import {
  GEL_SPACING_DBL,
  GEL_SPACING_TRPL,
  GEL_SPACING_QUAD,
  GEL_SPACING_QUIN,
  GEL_SPACING_SEXT,
  GEL_SPACING_SEPT,
} from '@bbc/gel-foundations/spacings';

// Services with fonts that have glyphs thinner than the majority of other fonts.
// This was mainly based on the old overrides (ie. Any group0 value < 2rem).
export const thinFontServices = [
  'pashto',
  'persian',
  'tamil',
  'telegu',
  'turkce',
  'urdu',
];

// If numberOfItems < 10, no extra spacing needs to be accounted for.
export const singleDigitDefault = {
  group0: GEL_SPACING_TRPL,
  group1: GEL_SPACING_TRPL,
  group2: GEL_SPACING_QUAD,
  group3: GEL_SPACING_QUIN,
  group5: GEL_SPACING_QUAD,
};

export const doubleDigitDefault = {
  group0: GEL_SPACING_QUIN,
  group1: GEL_SPACING_QUIN,
  group2: GEL_SPACING_SEXT,
  group3: GEL_SPACING_QUIN,
  group5: '3.9rem',
  group0_column: GEL_SPACING_QUAD,
  group1_column: GEL_SPACING_QUIN,
  group2_column: GEL_SPACING_SEXT,
  group3_5_column: '4rem',
};

export const singleDigitThin = {
  group0: GEL_SPACING_DBL,
  group1: GEL_SPACING_TRPL,
  group2: GEL_SPACING_TRPL,
  group3: GEL_SPACING_QUAD,
  group5: GEL_SPACING_QUAD,
};

export const doubleDigitThin = {
  group0: GEL_SPACING_DBL,
  group1: GEL_SPACING_TRPL,
  group2: GEL_SPACING_TRPL,
  group3: GEL_SPACING_TRPL,
  group5: GEL_SPACING_QUIN,
  group0_column: GEL_SPACING_QUAD,
  group1_column: GEL_SPACING_QUIN,
  group2_column: GEL_SPACING_SEXT,
  group3_5_column: GEL_SPACING_SEPT,
};
