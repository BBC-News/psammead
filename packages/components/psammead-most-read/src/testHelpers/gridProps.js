const mostReadListGridProps = {
  enableGelGutters: true,
  enableGelMargins: true,
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 6,
    group4: 8,
    group5: 20,
  },
};

const mostReadItemGridProps = {
  item: true,
  columns: {
    group0: 6,
    group1: 6,
    group2: 3,
    group3: 3,
    group4: 4,
    group5: 4,
  },
};

export { mostReadListGridProps, mostReadItemGridProps };
