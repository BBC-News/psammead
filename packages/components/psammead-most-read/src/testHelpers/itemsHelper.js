import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';

export const getItem = service => {
  const baseUrl = 'https://www.bbc.com';
  const { text, articlePath } = TEXT_VARIANTS[service];
  return {
    title: text,
    href: `${baseUrl}${articlePath}`,
  };
};

export const getItems = (service = 'news') => Array(10).fill(getItem(service));
