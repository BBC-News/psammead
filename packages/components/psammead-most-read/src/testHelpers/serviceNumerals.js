import {
  Burmese,
  Bengali,
  EasternArabic,
  Nepali,
  WesternArabic,
} from '@bbc/psammead-locales/numerals';

const serviceNumerals = service => {
  const servicesNonWesternNumerals = {
    arabic: EasternArabic,
    bengali: Bengali,
    burmese: Burmese,
    nepali: Nepali,
    pashto: EasternArabic,
    persian: EasternArabic,
    urdu: EasternArabic,
  };
  return servicesNonWesternNumerals[service]
    ? servicesNonWesternNumerals[service]
    : WesternArabic;
};

export default serviceNumerals;
