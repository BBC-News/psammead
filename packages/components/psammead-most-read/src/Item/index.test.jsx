import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin, arabic } from '@bbc/gel-foundations/scripts';
import { MostReadLink } from '.';
import { getItem, getItemWrapperArray } from '../testHelpers';

describe('MostReadLink', () => {
  shouldMatchSnapshot(
    'should render ltr correctly',
    <MostReadLink
      href={getItem('news').href}
      service="news"
      script={latin}
      title={getItem('news').title}
    />,
  );

  shouldMatchSnapshot(
    'should render rtl correctly',
    <MostReadLink
      dir="rtl"
      href={getItem('persian').href}
      service="persian"
      script={arabic}
      title={getItem('persian').title}
    />,
  );

  shouldMatchSnapshot(
    'should render with last updated date correctly',
    <MostReadLink
      href={getItem('news').href}
      service="news"
      script={latin}
      title={getItem('news').title}
    >
      {getItem('news', true).timestamp}
    </MostReadLink>,
  );
});

describe('MostReadItemWrapper', () => {
  shouldMatchSnapshot(
    'should render ltr correctly with 10 items',
    getItemWrapperArray({
      numberOfItems: 10,
      service: 'news',
      script: latin,
    }).map(item => item),
  );

  shouldMatchSnapshot(
    'should render rtl correctly with 10 items',
    getItemWrapperArray({
      numberOfItems: 10,
      service: 'persian',
      script: arabic,
      dir: 'rtl',
    }).map(item => item),
  );
});
