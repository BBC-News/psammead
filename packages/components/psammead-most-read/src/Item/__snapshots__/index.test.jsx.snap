// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostReadItemWrapper should render ltr correctly with 10 items 1`] = `
.c6 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:hover,
.c6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c5 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0;
  padding: 0;
}

.c1 {
  position: relative;
}

.c4 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-style: normal;
  font-size: 2rem;
  line-height: 2.25rem;
  position: relative;
  color: #B80000;
  margin: 0;
  padding: 0;
}

@media (max-width:14.9375rem) {
  .c0 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c0 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c0 {
    width: calc(20%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c0 {
    display: block;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c5 {
    padding-right: 0;
  }
}

@media (max-width:14.9375rem) {
  .c3 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c3 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c3 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c3 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c3 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c7 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c7 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c7 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c7 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c7 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c8 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c8 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c8 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c8 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c8 {
    min-width: 4.25rem;
  }
}

@media (max-width:14.9375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c9 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c9 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c10 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 2.5rem;
    line-height: 2.75rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 3.5rem;
    line-height: 3.75rem;
  }
}

<div>
  <li
    aria-labelledby="1 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        dir="ltr"
      >
        <span
          class="c4"
        >
          1
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="2 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c7"
        dir="ltr"
      >
        <span
          class="c4"
        >
          2
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="3 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        dir="ltr"
      >
        <span
          class="c4"
        >
          3
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="4 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c7"
        dir="ltr"
      >
        <span
          class="c4"
        >
          4
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="5 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c8"
        dir="ltr"
      >
        <span
          class="c4"
        >
          5
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="6 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c9"
        dir="ltr"
      >
        <span
          class="c4"
        >
          6
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="7 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c10"
        dir="ltr"
      >
        <span
          class="c4"
        >
          7
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="8 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c9"
        dir="ltr"
      >
        <span
          class="c4"
        >
          8
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="9 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c10"
        dir="ltr"
      >
        <span
          class="c4"
        >
          9
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="10 Could a computer ever create better art than a human?"
    class="c0 c1"
    dir="ltr"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c9"
        dir="ltr"
      >
        <span
          class="c4"
        >
          10
        </span>
      </div>
      <div
        class="c5"
        dir="ltr"
      >
        <a
          class="c6"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
</div>
`;

exports[`MostReadItemWrapper should render rtl correctly with 10 items 1`] = `
.c6 {
  font-size: 1.5rem;
  line-height: 2rem;
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:hover,
.c6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c5 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0;
  padding: 0;
}

.c1 {
  position: relative;
}

.c4 {
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 2rem;
  line-height: 2.25rem;
  position: relative;
  color: #B80000;
  margin: 0;
  padding: 0;
}

@media (max-width:14.9375rem) {
  .c0 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c0 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c0 {
    width: calc(20%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c0 {
    display: block;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 1.625rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 1.75rem;
    line-height: 2.25rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c5 {
    padding-left: 0;
  }
}

@media (max-width:14.9375rem) {
  .c3 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c3 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c3 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c3 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c3 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c7 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c7 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c7 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c7 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c7 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c8 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c8 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c8 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c8 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c8 {
    min-width: 4.25rem;
  }
}

@media (max-width:14.9375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c9 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c9 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c10 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 2.5rem;
    line-height: 2.75rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 3.5rem;
    line-height: 3.75rem;
  }
}

<div>
  <li
    aria-labelledby="۱ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۱
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="۲ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c7"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۲
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="۳ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۳
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="۴ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c7"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۴
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="۵ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c8"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۵
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="۶ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c9"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۶
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="۷ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c10"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۷
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="۸ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c9"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۸
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="۹ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c10"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۹
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    aria-labelledby="۱۰ در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد"
    class="c0 c1"
    dir="rtl"
    role="listitem"
  >
    <div
      class="c2"
    >
      <div
        class="c9"
        dir="rtl"
      >
        <span
          class="c4"
        >
          ۱۰
        </span>
      </div>
      <div
        class="c5"
        dir="rtl"
      >
        <a
          class="c6"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
</div>
`;

exports[`MostReadLink should render ltr correctly 1`] = `
.c1 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c0 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c0 {
    padding-right: 0;
  }
}

<div
  class="c0"
  dir="ltr"
>
  <a
    class="c1"
    href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
  >
    Could a computer ever create better art than a human?
  </a>
</div>
`;

exports[`MostReadLink should render rtl correctly 1`] = `
.c1 {
  font-size: 1.5rem;
  line-height: 2rem;
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c0 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 1rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1.625rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2.25rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c0 {
    padding-left: 0;
  }
}

<div
  class="c0"
  dir="rtl"
>
  <a
    class="c1"
    href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
  >
    لماذا يخجل البعض من اسم قريته في مصر؟
  </a>
</div>
`;

exports[`MostReadLink should render with last updated date correctly 1`] = `
.c1 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c0 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c2 {
  font-size: 0.875rem;
  line-height: 1.125rem;
  color: #6E6E73;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c0 {
    padding-right: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

<div
  class="c0"
  dir="ltr"
>
  <a
    class="c1"
    href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
  >
    Could a computer ever create better art than a human?
  </a>
  <time
    class="c2"
    datetime="2019-03-01T14:00+00:00"
  >
    Last updated: 5th November 2016
  </time>
</div>
`;
